File: Android/app/src/main/java/com/lunesu/pengchauferry/ui/ferry/Strings.kt
package com.lunesu.pengchauferry.ui.ferry

import android.content.res.Resources
import com.lunesu.pengchauferry.FerryDay
import com.lunesu.pengchauferry.FerryPier
import com.lunesu.pengchauferry.R

object Strings {
    val PIERS = mapOf(
        FerryPier.Central to R.string.Central,
        FerryPier.PengChau to R.string.PengChau,
        FerryPier.TrappistMonastery to R.string.TrappistMonastery,
        FerryPier.DiscoveryBay to R.string.DiscoveryBay,
        FerryPier.MuiWo to R.string.MuiWo,
        FerryPier.CheungChau to R.string.CheungChau,
        FerryPier.ChiMaWan to R.string.ChiMaWan,
        FerryPier.HeiLingChau to R.string.HeiLingChau
    )

    val PIERS_DUAL = mapOf(
        FerryPier.Central to R.string.Central2,
        FerryPier.PengChau to R.string.PengChau2,
        FerryPier.TrappistMonastery to R.string.TrappistMonastery2,
        FerryPier.DiscoveryBay to R.string.DiscoveryBay2,
        FerryPier.MuiWo to R.string.MuiWo2,
        FerryPier.CheungChau to R.string.CheungChau2,
        FerryPier.ChiMaWan to R.string.ChiMaWan2,
        FerryPier.HeiLingChau to R.string.HeiLingChau2
    )

    val DAYS = mapOf(
        FerryDay.Monday to R.string.monday,
        FerryDay.Tuesday to R.string.tuesday,
        FerryDay.Wednesday to R.string.wednesday,
        FerryDay.Thursday to R.string.thursday,
        FerryDay.Friday to R.string.friday,
        FerryDay.Saturday to R.string.saturday,
        FerryDay.Sunday to R.string.sunday,
        FerryDay.Holiday to R.string.holiday
    )

    fun localized(pier: FerryPier, resources: Resources) : String =
        resources.getString(PIERS.getValue(pier))

    fun localized(day: FerryDay, resources: Resources) : String =
        resources.getString(DAYS.getValue(day))
}

File: iOS/PengChauFerry/ui/Strings.swift
//
//  Strings.swift
//  PengChauFerry
//
//  Created by Lionello Lunesu on 7/3/2020.
//  Copyright © 2020 Lionello Lunesu. All rights reserved.
//

import SwiftUI

class Strings {
    static let PIERS : [FerryPier?:LocalizedStringKey] = [
        .Central : "中環 Central",
        .PengChau : "坪洲 Peng Chau",
        .TrappistMonastery : "聖母神樂院 Trappist Monastery",
        .DiscoveryBay : "愉景灣 Discovery Bay",
        .MuiWo : "梅窩 Mui Wo",
        .CheungChau : "長洲 Cheung Chau",
        .ChiMaWan : "芝麻灣 Chi Ma Wan",
        .HeiLingChau : "喜靈洲 HeiLingChau"
    ]

    static let DAYS : [FerryDay?:LocalizedStringKey] = [
        .Monday : "Monday",
        .Tuesday : "Tuesday",
        .Wednesday : "Wednesday",
        .Thursday : "Thursday",
        .Friday : "Friday",
        .Saturday : "Saturday",
        .Sunday : "Sunday",
        .Holiday : "Holiday"
    ]

    static func localized(_ pier: FerryPier, _ bundle: Bundle) -> String {
        bundle.localizedString(forKey: pier.rawValue, value: nil, table: nil)
    }

    static func localized(_ day: FerryDay, _ bundle: Bundle) -> String {
        bundle.localizedString(forKey: day.toString(), value: nil, table: nil)
    }

}
